### PR TITLE
Xfemm group numbers and block labels

### PR DESCRIPTION
**Pull request by Anvar-Khamitov**
This pull request solves an issue #88.

### CODE REVIEW

File: model_obj/Component.m [hunk 1]
```diff
@@ -15,15 +15,17 @@
...
             validateattributes(maker,  {'MakerBase'},  {'nonempty'});
             
             for i = 1:length(obj.crossSections)
                 cs(i) = obj.crossSections(i).draw(drawer);                
             end
             
-            obj.makeSolid.run(obj.name, obj.material.name, cs, maker)
+            tok2 = obj.makeSolid.run(obj.name, obj.material.name, cs, maker);
+            
+            rettok = {cs, tok2};
```
Comment (elsevers) [inline model_obj/Component.m]: Create a class the give a contract for the contents of `rettok`.

File: model_obj/makeSolid/MakeExtrude.m [hunk 2]
```diff
@@ -14,22 +14,23 @@
...
             end
             
             %2. Make via extrusion
-            maker.extrude(name, material, obj.dim_depth);
+            tok2 = maker.extrude(name, material, obj.dim_depth, tok1);
             
             %3. TO DO: Move to final location
             %maker.Move(name, obj.location);
             
+            tok = {tok1, tok2};
```
Comment (elsevers) [inline model_obj/makeSolid/MakeExtrude.m]: Create a class for `tok` to guarantee the fields

File: tools/xfemm/XFEMM.m [hunk 3]
```diff
@@ -387,22 +387,28 @@ function plot(obj)
...
+            endxy = double(endxy);
             [obj.FemmProblem, nodeinds, nodeids] = addnodes_mfemm(...
                 obj.FemmProblem, [startxy(1); endxy(1)],...
                 [startxy(2); endxy(2)]);
             
             [obj.FemmProblem, seginds] = addsegments_mfemm(obj.FemmProblem,...
                 nodeids(1), nodeids(2));
             
-            line = seginds;
+            line = {seginds;0};
```
Comment (elsevers) [inline tools/xfemm/XFEMM.m]: Create classes to compliment the XFEMM tool that specify these tokens. The class should have a couple of fields, for example:

Class: drawToken
Fields: segmentIndices, geometryType

File: tools/xfemm/XFEMM.m [hunk 4]
```diff
@@ -452,18 +454,73 @@ function plot(obj)
...
+            
+            obj.FemmProblem = addmaterials_mfemm(obj.FemmProblem, material);
+            obj.FemmProblem = addblocklabel_mfemm(obj.FemmProblem, ...
+                innerCoord(1), innerCoord(2), ...
+                                   'BlockType', material);
+            tok2 = [];
+        end
+        % Save information about inner coordinates of the cross-section
+        function innerCoord = prepareSection(obj,csToken)
+            innerCoord = double(csToken.innerCoord);
```
Comment (elsevers) [inline tools/xfemm/XFEMM.m:493]: This should return `csToken.innerCoord` without converting the units

File: tools/xfemm/XFEMM.m [hunk 5]
```diff
@@ -452,18 +454,73 @@ function plot(obj)
...
+            obj.FemmProblem = addblocklabel_mfemm(obj.FemmProblem, ...
+                innerCoord(1), innerCoord(2), ...
+                                   'BlockType', material);
+            tok2 = []; 
         end
-        
-        function extrude(obj, name, material, depth)
-
+        % Add block label and assign material
+        function tok2 = extrude(obj, name, material, depth, innerCoord)
```
Comment (elsevers) [inline tools/xfemm/XFEMM.m]: As documented in issue #66, this needs to convert units. It does not need to be done in this PR. Can be reserved for a future PR.

**Review by elsevers — CHANGES_REQUESTED** [review]

File: tools/xfemm/XFEMM.m (hunk 4 above)
Comment (elsevers) [reply tools/xfemm/XFEMM.m:493]: To be resolved by issue #66

**Review by elsevers — APPROVED** [review]
Looks good. I think this is ready to merge.

**Review by npetersen2 — APPROVED** [review]

**Pull request merged**